plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php

```diff
@@ -189,11 +195,22 @@ function handleRevisionElement($node, $stageId, $fileId) {
 
 		$submissionFile->setRevision($revisionId);
 
+		if ($sourceFileAndRevision) {
+			$sourceFileId = $deployment->getFileDBId($sourceFileAndRevision[0], $sourceFileAndRevision[1]);
+			if ($sourceFileId) {
+				$submissionFile->setSourceFileId($sourceFileId);
+				$submissionFile->setSourceRevision($sourceFileAndRevision[1]);
+			}
+		}
+
 		if ($errorOccured) {
 			// if error occured, the file cannot be inserted into DB, becase
 			// genre, uploader and user group are required (e.g. at name generation).
 			$submissionFile = null;
 		} else {
+			// if the same file is already inserted, take its DB file ID
+			$DBId = $deployment->getFileDBId($fileId);
```

Comment:
I realized that this is not right -- we should be able to a) look if a revision is already inserted and b) to look only after the new file id. In case a) we have to provide revisionId parameter and the only matching value should be returned and if not found null. In case of b) we do not care about revisionID, we only want the new DB file ID. OK? The final changes consider all this...